src/utils/logging.rs

```diff
@@ -0,0 +1,58 @@
+use std::{
+    fs,
+    io::{self, prelude::*},
+    os::unix::io::RawFd,
+    path::PathBuf,
+};
+
+use crate::utils::consts::{MOSAIC_TMP_LOG_DIR, MOSAIC_TMP_LOG_FILE};
+
+pub fn _debug_log_to_file(message: String) -> io::Result<()> {
```

Comment:
I think if these functions are public we can remove all the leading underscores and not get warnings, right?